### PR TITLE
build: add commit hash to Config at build for proper discovery readme

### PR DESCRIPTION
**Pull request by chofnar**
**What type of PR is this?**
documentation I think

**Which issue does this PR fix**:
#852

**What does this PR do / Why do we need it**:
This PR gets the commit hash and writes it into the new CommitHash variable in api.Config, which is used in the discovery api to point to the extensions readme link of the used build:

```
curl localhost:5000/v2/_oci/ext/discover
{"extensions":[{"name":"_zot","url":"https://github.com/project-zot/zot/blob/v1.4.3-rc2/pkg/extensions/_zot.md","description":"zot registry extension","endpoints":["/v2/_zot/ext/search"]}]}
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — APPROVED** [review]
LGTM

**Comment by rchincha** [comment]
not commit hash pls, use the tag

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export GO111MODULE=on
 TOP_LEVEL=$(shell git rev-parse --show-toplevel)
 COMMIT_HASH=$(shell git describe --always --tags --long)
+COMMIT_LONGHASH=$(shell git rev-parse HEAD)
```
Comment (andaaron) [inline Makefile:4]: Let's use the latest tag, ideally `git describe --tags --abbrev=0`.
But make sure this works for release builds. You may need to overwrite this in the github action using one of the environment variables: https://docs.github.com/en/actions/learn-github-actions/environment-variables

**Review by andaaron — DISMISSED** [review]
LGTM

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -153,6 +153,7 @@ jobs:
         with:
           file: 'stacker.yaml'
           build-args: |
+            RELEASE_TAG=&{{ github.event.release.tag_name }}
```
Comment (rchincha) [inline .github/workflows/ci-cd.yml:156]: $ not &

File: pkg/extensions/extension_search.go [hunk 3]
```diff
@@ -103,7 +103,7 @@ func GetExtensions(config *config.Config) distext.ExtensionList {
 	if config.Extensions != nil && config.Extensions.Search != nil {
 		endpoints := []string{constants.ExtSearchPrefix}
 		searchExt := getExtension("_zot",
-			"https://github.com/project-zot/zot/tree/main/pkg/extensions/_zot.md",
+			"https://github.com/project-zot/zot/blob/"+config.ReleaseTag+"/pkg/extensions/_zot.md",
 			"zot registry extension",
```
Comment (rchincha) [inline pkg/extensions/extension_search.go:107]: Unrelated to your PR, but also make this "zot registry extensions" <- plural

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (chofnar) [reply .github/workflows/ci-cd.yml]: Fixed

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -198,6 +199,7 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (chofnar) [reply .github/workflows/ci-cd.yml]: Fixed

File: pkg/extensions/extension_search.go (hunk 3 above)
Comment (chofnar) [reply pkg/extensions/extension_search.go]: Added as requested

**Review by andaaron — APPROVED** [review]

**Pull request merged**